### PR TITLE
[IMPROVE] Resize custom emojis on upload instead of saving at max res

### PR DESCRIPTION
**Pull request by KevLehman**
## Proposed changes (including videos or screenshots)

- Create new MediaService (ideally, should be in charge of all media-related operations)
- Resize emojis to 128x128

### CODE REVIEW

File: server/services/image/service.ts [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+		if (!keepType) {
+			transformer.jpeg();
+		}
+
+		if (blur) {
+			transformer.blur();
+		}
+
+		return transformer.toBuffer({ resolveWithObject: true }).then(({ data, info: { width, height } }) => ({ data, width, height }));
```
Comment (sampaiodiego) [inline server/services/image/service.ts:46]: let me be a bit picky here 🙈 

can we avoid using `when` whenever possible? 😬 IMO the code using `await` is better for read, so it'd look like something like this (in this example, might be better to rename params so doesn't need to rename the result):

```javascript
		const { 
			data, 
			info: { 
				width: widthInfo, 
				height: heightInfo,
			},
		} = await transformer.toBuffer({ resolveWithObject: true });

		return {
			data,
			width: widthInfo,
			height: heightInfo,
		};

File: server/services/image/service.ts [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+		if (!keepType) {
+			transformer.jpeg();
+		}
+
+		if (blur) {
+			transformer.blur();
+		}
+
+		const result = transformer.toBuffer({ resolveWithObject: true }).then(({ data, info: { width, height } }) => ({ data, width, height }));
```
Comment (sampaiodiego) [inline server/services/image/service.ts:61]: I'd use `await` here as well 😬

**Review by sampaiodiego — COMMENTED** [review]
Nice idea having a service for this 👏 

to make use of it's full potential though we need to make sure it works when using a network transporter (when using microservices on EE).. looks like moleculer supports ot (https://moleculer.services/docs/0.14/actions.html#Streaming) , so might be just a matter of testing it out :D

**Comment by KevLehman** [comment]
I haven't tried running it as a ee microservice, but yeah, moleculer supports streams/buffers to be sent so we shouldn't have a big issue 😬 

The other comments were resolved too.

**Review by sampaiodiego — APPROVED** [review]

**Comment by sampaiodiego** [comment]
I'm sorry @antoine-roux this got hidden in the release notes, but we have increased the emoji size to 512x512 [here](https://github.com/RocketChat/Rocket.Chat/pull/22044/files#diff-af40e039878bf836e63c93a83cc0c18b73eaffcfed8b1890db7922a15b336f4bR43).. hope that works for everyone =)

**Pull request merged**